### PR TITLE
Fixes #44237: Add `columnNumber` variable

### LINKED ISSUE #44237: Add variable to tasks.json for cursor column number
- VSCode Version: 1.20.1
- OS Version: All

The [variables available](https://code.visualstudio.com/docs/editor/variables-reference) for tasks.json are very helpful. Unfortunately, while the line number is available in ${lineNumber}, there's no line column variable. 

The column number would be helpful for me because I have a script which can perform different actions based on where the cursor is.  If the cursor is on a function name, for example, it can launch a documentation program (outside of vscode) to show me the function documentation.  But I need both the cursor line and column passed to the script so it can determine (or at least guess) in what type of context the cursor is found.

Is there a way we could get a ${column} value to return the column number for use in tasks?  Thanks!

### PR DESCRIPTION
**Pull request by keithel**
This adds a new variable that you can access from tasks and other places to be able to substitute in the currently selected column number from the active editor.

### CODE REVIEW

**Comment by keithel** [comment]
@microsoft-github-policy-service agree

**Comment by keithel** [comment]
@alexr00 is there a way I can advocate/lobby for community approvals to my PR?

File: src/vs/workbench/api/common/extHostVariableResolverService.ts [hunk 1]
```diff
@@ -100,20 +107,24 @@ class ExtHostVariableResolverService extends AbstractVariableResolverService {
...
+				const selection = getSelection(editorService);
+				if (selection) {
+					return String(selection.end.line + 1);
+				}
+				return undefined;
+			},
+			getColumnNumber: (): string | undefined => {
+				const selection = getSelection(editorService);
+				if (selection) {
+					return String(selection.end.character + 1);
```
Comment (alexr00) [inline src/vs/workbench/api/common/extHostVariableResolverService.ts:127]: Just getting the end character doesn't actually give the column if there are characters that span more than one column, like tabs.

**Review by alexr00 — CHANGES_REQUESTED** [review]
Thanks for the PR! I've reduced some code duplication that was made more obvious by this change. It looks good, but it doesn't actually work for tasks. I'm not sure we expose something on the extension host that will solve this.

**Comment by alexr00** [comment]
Closing in favor of https://github.com/microsoft/vscode/pull/65264

**Comment by keithel** [comment]
Thank you so much for this @alexr00 !
Your work is very much appreciated.

**Comment by keithel** [comment]
I don't suppose there's a way I can "buy you a coffee" or whatnot? (donate to a charity of your choice?, something else?)

**Comment by alexr00** [comment]
Nope, thank you, but it's always nice to hear that a change is genuinely useful to folks!

**Pull request closed without merging**